src/duration/mod.rs

```diff
@@ -56,8 +57,24 @@ impl Display for DurationError {
 /// Can be deserialized from duration strings
 /// of the format `"{number} {period}"`,
 /// e.g. `"1 hour"` or `"10 minutes"`.
-#[derive(Debug, PartialEq)]
-pub struct Duration(u64);
+#[derive(Clone, Debug, Default, Serialize, PartialEq)]
+pub struct Duration(pub u64);
```

Comment:
Duration was the only one that was not a string and also the only one that already had a type and didn't use the`validate` module. This is why I didn't use the macro to create it's type.

Also I implemented deserialize here to keep it close to the `struct` declaration.